app/controllers/filter.js

```diff
@@ -0,0 +1,14 @@
+const User = require("../models/User.js");
+
+async function filter(req, res, next) {
+  try {
+    const users = await User.find();
+    const filteredUsers = users.filter(user => user.gender === user.input);
```

Comment:
You should await the users.filter otherwise the program will run further and won't filter results because it logs earlier then it is done filtering
```suggestion
    const filteredUsers = await users.filter(user => user.gender === user.input);
```